src/nvim/window.c

```diff
@@ -5235,30 +5235,22 @@ void win_free(win_T *wp, tabpage_T *tp)
   // Remove the window from the b_wininfo lists, it may happen that the
   // freed memory is re-used for another window.
   FOR_ALL_BUFFERS(buf) {
-    for (wininfo_T *wip = buf->b_wininfo; wip != NULL; wip = wip->wi_next) {
+    WinInfo *wip_wp = NULL;
+    size_t pos_null = kv_size(buf->b_wininfo);
+    for (size_t i = 0; i < kv_size(buf->b_wininfo); i++) {
+      WinInfo *wip = kv_A(buf->b_wininfo, i);
       if (wip->wi_win == wp) {
-        wininfo_T *wip2;
-
-        // If there already is an entry with "wi_win" set to NULL it
-        // must be removed, it would never be used.
```

Comment:
Why is this comment removed?

Reply:
I think it should be moved above the `pos_null < kv_size(buf->b_wininfo)` check.